flink-java/src/main/java/org/apache/flink/api/java/utils/ParameterTool.java

```diff
@@ -560,9 +591,13 @@ protected Object clone() throws CloneNotSupportedException {
 	 * @return The Merged {@link ParameterTool}
 	 */
 	public ParameterTool mergeWith(ParameterTool other) {
-		ParameterTool ret = new ParameterTool(this.data);
-		ret.data.putAll(other.data);
-		ret.unrequestedParameters.addAll(other.unrequestedParameters);
+		Map<String, String> resultData = new HashMap<>(data.size() + other.data.size());
+		resultData.putAll(data);
+		resultData.putAll(other.data);
+
+		ParameterTool ret = new ParameterTool(resultData);
+
+		ret.unrequestedParameters.putAll(other.unrequestedParameters);
```

Comment:
what about `this.unrequestedParameters`?

Reply:
Forgot to add it.